clean.cmd

```diff
@@ -0,0 +1,14 @@
+@if not defined _echo @echo off
+
+git submodule foreach --recursive git clean -xdf
+git submodule foreach --recursive git reset --hard
+
+if [%1] == [-a] (
+  call git clean -xdf
+  exit /b %ERRORLEVEL%
+)
+
+if EXIST "%~dp0Tools\dotnetcli\dotnet.exe" (
```

Comment:
Seems like this should go before the `git clean -xdf` at the root.  If you pass in `-a`, this will never exist since it will get cleaned above.

Reply:
That is my intention. If you do "-a" it will exit after doing the full git clean. I did the exist check here to make handle the case where you have a clean environment and you call "clean" (not -a) but there isn't anything to clean and the cli doesn't exist.